docs/next/components/mdx/MDXComponents.tsx

```diff
@@ -338,6 +339,11 @@ const Experimental = () => {
 };
 
 const Pre = ({children, ...props}) => {
+  // Retrieve the DAG image to display next to the code block, if any
+  const childrenArr = React.Children.toArray(children);
+  const preComponent = childrenArr.length > 0 ? childrenArr[0] : null;
+  const dagImage = preComponent ? preComponent['props']['dagimage'] : null;
```

Comment:
It's pretty hacky to reach into the props of a child component -- could you instead provide `dagImage` as a separate prop, and leave `children` as is?

Reply:
Yeah, I'd like to resolve this. I'm not confident in the process in which MDX is being converted into React components - somehow the code block is being transformed into a `pre` tag, which is then being wrapped into our `Pre` component. Ideally instead we would just convert the `pre` tag or be able to move some of its props to our `Pre` component.

Reply:
Alright... I found /a/ solution: https://github.com/dagster-io/dagster/pull/9714/commits/860d0228c72c46222fd13866e5f7023bd98d699d

It's a bit gross, but I think it's the nicest solution we'll get.

Right now, a code block in MDX (with three backticks) is converted into a <code> tag wrapped in a <pre> tag, which are nicely styled. However, the <code> tag contains all of the props (e.g. the dag image).

The change in the above commit takes the generated <pre> tag and no-ops the styling. Then, it creates a new <pre> tag in our <code> tag replacement, once we know whether the dag image is set or not.